test/framework/src/main/java/org/opensearch/test/client/RandomizingClient.java

```diff
@@ -61,7 +62,7 @@ public RandomizingClient(Client client, Random random) {
         // given that they return `size*num_shards` hits instead of `size`
         defaultSearchType = RandomPicks.randomFrom(random, Arrays.asList(SearchType.DFS_QUERY_THEN_FETCH, SearchType.QUERY_THEN_FETCH));
         if (random.nextInt(10) == 0) {
-            defaultPreference = Preference.LOCAL.type();
+            defaultPreference = RandomPicks.randomFrom(random, EnumSet.of(Preference.PRIMARY_FIRST, Preference.LOCAL)).type();
```

Comment:
Wonder if we should add the whole set of preferences here instead?

Reply:
I have added'em in for now. Lets see how the tests behave.

Reply:
That did not fare well. The client is used across all the tests - I think it created havoc since different tests might test for different features, some of which might not align with the preferences. Reverted the change.